File: direct-file/backend/src/main/java/gov/irs/directfile/api/stateapi/domain/StateProfile.java
package gov.irs.directfile.api.stateapi.domain;

import java.util.List;
import java.util.Map;

import gov.irs.directfile.models.StateOrProvince;

public record StateProfile(
        StateOrProvince stateCode,
        String taxSystemName,
        String landingUrl,
        String defaultRedirectUrl,
        String departmentOfRevenueUrl,
        String filingRequirementsUrl,
        String transferCancelUrl,
        String waitingForAcceptanceCancelUrl,
        List<String> redirectUrls,
        /*
         A map with keys that correspond to DF-understandable language codes to the corresponding codes
         used by the state. Maps back to state-api's StateLanguage entity e.g:

         {
             en: `en`
             es: `es `
         }

         OR

         {
             en: `english`
             es: `spanish`
         }
        */
        Map<String, String> languages,
        String customFilingDeadline,
        Boolean acceptedOnly) {}

File: direct-file/df-client/df-client-app/src/types/StateProfile.ts
import { StateOrProvince } from './StateOrProvince.js';

// Align with state-tax StateProfile
export type StateProfile = {
  stateCode: StateOrProvince;
  landingUrl: string;
  defaultRedirectUrl: string | null;
  departmentOfRevenueUrl: string | null;
  filingRequirementsUrl: string | null;
  transferCancelUrl: string | null;
  waitingForAcceptanceCancelUrl: string | null;
  redirectUrls: string[];
  languages: Record<string, string>;
  taxSystemName: string;
  acceptedOnly: boolean;
  // customFilingDeadline is timezone agnostic.
  // e.g. A value of 04-17 23:59:59.999999 is 04-17 23:59:59.999999 regardless of the user's timezone
  customFilingDeadline: string | null;
};

export type StateProfileResponse = {
  stateProfile: StateProfile | null;
  error: string | null;
};
